### ISSUE TITLE
Issue in istio-pilot-0: hook failed "install"

### ISSUE THREAD

**Issue by amitk75** [canonical/bundle-kubeflow#509]
Hi,

I have deployed kubeflow 1,6 using juju on microk8s 1.25 on my laptop running Ubuntu 22.04. The istio-pilot pod is showing following errors. Other services are in waiting state probably because this issue (showing status: List of ingress versions not found for apps: istio-pilot)

Can you please help me with this issue?

```
2022-10-08 12:22:41 INFO juju.cmd supercommand.go:56 running containerAgent [2.9.34 90e2f047763059f0b8a57941ae0907346464aee8 gc go1.19]
starting containeragent unit command
containeragent unit "unit-istio-pilot-0" start (2.9.34 [gc])
2022-10-08 12:22:41 INFO juju.cmd.containeragent.unit runner.go:556 start "unit"
2022-10-08 12:22:41 INFO juju.worker.upgradesteps worker.go:60 upgrade steps for 2.9.34 have already been run.
2022-10-08 12:22:41 INFO juju.worker.probehttpserver server.go:157 starting http server on [::]:3856
2022-10-08 12:22:41 INFO juju.api apiclient.go:1055 cannot resolve "controller-service.controller-myk8s-localhost.svc.cluster.local": lookup controller-service.controller-myk8s-localhost.svc.cluster.local: operation was canceled
2022-10-08 12:22:41 INFO juju.api apiclient.go:688 connection established to "wss://10.152.183.127:17070/model/3d42a6b1-a24a-4ecc-8bc5-bf071fffd593/api"
2022-10-08 12:22:41 INFO juju.worker.apicaller connect.go:163 [3d42a6] "unit-istio-pilot-0" successfully connected to "10.152.183.127:17070"
2022-10-08 12:22:41 INFO juju.worker.migrationminion worker.go:142 migration phase is now: NONE
2022-10-08 12:22:41 INFO juju.worker.logger logger.go:120 logger worker started
2022-10-08 12:22:41 INFO juju.worker.leadership tracker.go:194 istio-pilot/0 promoted to leadership of istio-pilot
2022-10-08 12:22:42 WARNING juju.worker.proxyupdater proxyupdater.go:282 unable to set snap core settings [proxy.http= proxy.https= proxy.store=]: exec: "snap": executable file not found in $PATH, output: ""
2022-10-08 12:22:42 INFO juju.agent.tools symlinks.go:20 ensure jujuc symlinks in /var/lib/juju/tools/unit-istio-pilot-0
2022-10-08 12:22:42 INFO juju.worker.caasupgrader upgrader.go:113 abort check blocked until version event received
2022-10-08 12:22:42 INFO juju.worker.caasupgrader upgrader.go:119 unblocking abort check
2022-10-08 12:22:42 INFO juju.worker.uniter uniter.go:326 unit "istio-pilot/0" started
2022-10-08 12:22:42 INFO juju.worker.uniter uniter.go:344 hooks are retried true
2022-10-08 12:22:42 INFO juju.worker.uniter resolver.go:145 awaiting error resolution for "install" hook
2022-10-08 12:22:47 INFO juju.worker.uniter resolver.go:145 awaiting error resolution for "install" hook
2022-10-08 12:22:48 INFO juju-log Running legacy hooks/install.
2022-10-08 12:22:51 WARNING install 
2022-10-08 12:22:53 WARNING install - Processing resources for Istio core.
2022-10-08 12:22:53 WARNING install ✔ Istio core installed
2022-10-08 12:22:54 WARNING install - Processing resources for Istiod.2022-10-08T12:22:54.590378Z	error	installer	failed to update resource with server-side apply for obj PodDisruptionBudget/kubeflow/istiod: no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"
2022-10-08 12:22:54 WARNING install 2022-10-08T12:22:54.680241Z	error	installer	failed to update resource with server-side apply for obj HorizontalPodAutoscaler/kubeflow/istiod: no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta1"
2022-10-08 12:22:54 WARNING install 
2022-10-08 12:22:54 WARNING install ✘ Istiod encountered an error: failed to update resource with server-side apply for obj PodDisruptionBudget/kubeflow/istiod: no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"
2022-10-08 12:22:54 WARNING install failed to update resource with server-side apply for obj HorizontalPodAutoscaler/kubeflow/istiod: no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta1"
2022-10-08 12:22:55 WARNING install - Pruning removed resources2022-10-08T12:22:55.380478Z	warn	installer	retrieving resources to prune type autoscaling/v2beta1, Kind=HorizontalPodAutoscaler: no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta1" not found
2022-10-08 12:22:56 WARNING install 2022-10-08T12:22:56.304776Z	warn	installer	retrieving resources to prune type policy/v1beta1, Kind=PodDisruptionBudget: no matches for kind "PodDisruptionBudget" in version "policy/v1beta1" not found
2022-10-08 12:22:59 WARNING install Error: failed to install manifests: errors occurred during operation
2022-10-08 12:22:59 ERROR juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "./src/charm.py", line 333, in <module>
    main(Operator)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/main.py", line 424, in main
    framework.reemit()
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/framework.py", line 746, in reemit
    self._reemit()
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/framework.py", line 783, in _reemit
    custom_handler(event)
  File "./src/charm.py", line 72, in install
    subprocess.check_call(
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['./istioctl', 'install', '-y', '-s', 'profile=minimal', '-s', 'values.global.istioNamespace=kubeflow']' returned non-zero exit status 1.
2022-10-08 12:23:00 ERROR juju.worker.uniter.operation runhook.go:140 hook "install" (via hook dispatching script: dispatch) failed: exit status 1

```

**Comment by misohu** [comment]
Hi @amitk75 charmed kubeflow currently supports kubernetes 1.22. You can reinstall your microk8s for 1.22 and redeploy.
```
sudo snap remove microk8s --purge; juju unregister -y <controller-name>
sudo snap install microk8s --channel=1.22/stable --classic
```

**Comment by ca-scribner** [comment]
Adding a few more details, Kubeflow itself officially supports k8s 1.22.  We (**Charmed** Kubeflow) believe it should work on 1.23 and 1.24 as well, but know that it will not work on 1.25 right now.  At a minimum, we know the version of istio we're using does not support 1.25 (for that, we need Istio 1.15).  We're pushing to make sure the next release of upstream Kubeflow supports k8s 1.25

**Comment by amitk75** [comment]
Thank you all! I have installed Microk8s 1.22 and Kubeflow. 
Sent from Yahoo Mail on Android 
 
  On Thu, 13 Oct 2022 at 7:14 pm, Andrew ***@***.***> wrote:   


Adding a few more details, Kubeflow itself officially supports k8s 1.22. We (Charmed Kubeflow) believe it should work on 1.23 and 1.24 as well, but know that it will not work on 1.25 right now. At a minimum, we know the version of istio we're using does not support 1.25 (for that, we need Istio 1.15). We're pushing to make sure the next release of upstream Kubeflow supports k8s 1.25

**Comment by ca-scribner** [comment]
That's great to hear, thanks for letting us know!

**Comment by SandPhoenixX517** [comment]
This issue is happening even with Kubernetes 1.24. Is there a solution for this?

**Comment by aayushharwani-aidash** [comment]
Facing this issue with eks 1.24 and kubeflow 1.7, any solution for this ? @misohu @ca-scribner @SandPhoenixX517

**Comment by aayushharwani-aidash** [comment]
can we open this issue, as it hasn't been resolved yet.

**Comment by ca-scribner** [comment]
@aayushharwani-aidash if you're still facing this, can you open a new issue and add your logs?  This issue was caused specifically by k8s 1.25, so we don't want to mix everything together.
